pyprep/find_noisy_channels.py

```diff
@@ -0,0 +1,617 @@
+
+        # Make sure that we got an MNE object
+        assert isinstance(raw, mne.io.BaseRaw)
+
+        self.raw_mne = raw.copy()
+        self.EEGData = self.raw_mne.get_data(picks="eeg")
+        self.EEGData_beforeFilt = self.EEGData
+        self.ch_names_original = np.asarray(raw.info["ch_names"])
+        self.sample_rate = raw.info["sfreq"]
+        self.n_chans_original = len(self.ch_names_original)
+        self.n_chans_new = self.n_chans_original
+        self.signal_len = len(self.raw_mne.times)
+        self.original_dimensions = np.shape(self.EEGData)
+        self.new_dimensions = self.original_dimensions
+        self.original_channels = np.arange(self.original_dimensions[0])
+        self.new_channels = self.original_channels
+        self.ch_names_new = self.ch_names_original
+        self.channels_interpolate = self.original_channels
+
+        # The identified bad channels
+        self.bad_by_nan = []
+        self.bad_by_flat = []
+        self.bad_by_deviation = []
+        self.bad_by_hf_noise = []
+        self.bad_by_correlation = []
+        self.bad_by_SNR = []
+        self.bad_by_dropout = []
+        self.bad_by_ransac = []
+
+    def get_bads(self, verbose=False):
+
+        """Get a list of all bad channels.
+        Parameters
+        __________
+        verbose : boolean
+            If verbose, print a summary of bad channels.
+        """
+
+        bads = (
+            self.bad_by_nan
+            + self.bad_by_flat
+            + self.bad_by_deviation
+            + self.bad_by_hf_noise
+            + self.bad_by_SNR
+            + self.bad_by_correlation
+            + self.bad_by_dropout
+            + self.bad_by_ransac
+        )
+        bads = list(set(bads))
+
+        if verbose:
+            print("Found {} uniquely bad channels.".format(len(bads)))
+            print("\n{} by n/a: {}".format(len(self.bad_by_nan), self.bad_by_nan))
+            print("\n{} by flat: {}".format(len(self.bad_by_flat), self.bad_by_flat))
+            print(
+                "\n{} by deviation: {}".format(
+                    len(self.bad_by_deviation), self.bad_by_deviation
+                )
+            )
+            print(
+                "\n{} by hf noise: {}".format(
+                    len(self.bad_by_hf_noise), self.bad_by_hf_noise
+                )
+            )
+            print(
+                "\n{} by correl: {}".format(
+                    len(self.bad_by_correlation), self.bad_by_correlation
+                )
+            )
+            print("\n{} by SNR {}".format(len(self.bad_by_SNR), self.bad_by_SNR))
+            print(
+                "\n{} by dropout: {}".format(
+                    len(self.bad_by_dropout), self.bad_by_dropout
+                )
+            )
+            print(
+                "\n{} by ransac: {}".format(len(self.bad_by_ransac), self.bad_by_ransac)
+            )
+
+        return bads
+
+    def find_all_noisy_channels(self, ransac=True):
+
```

Comment:
Please change this function name to find_all_bads so I can easily compare your code with Noisydata